pinot-common/src/main/java/org/apache/pinot/common/request/context/predicate/RangePredicate.java

```diff
@@ -67,15 +69,17 @@ public RangePredicate(ExpressionContext lhs, String range) {
     int upperLength = upper.length();
     _upperInclusive = upper.charAt(upperLength - 1) == UPPER_INCLUSIVE;
     _upperBound = upper.substring(0, upperLength - 1);
+    _rangeDataType = null;
```

Comment:
Consider making this `UNKNOWN`? We should also put some comments to prevent people from using the data type if the predicate is constructed this way (which is the case most of the time).

Reply:
Ack